### ISSUE TITLE
User primary_group is not set initialy

### ISSUE THREAD

**Issue by Lefthandmedia** [modxcms/revolution#12766]
When adding a new user and selecting a usergroup for that user, the primary_group remains 0 on creation.
Only after adding a second usergroup the value for primary_group changes to the correct usergroupID
This behaviour is relative new. My application did not have this bug before. I think this is introduced in 2.4 but i'm not sure

this issue resembles #11139

**Comment by Kookabura** [comment]
I think it doesn't connected with #11139. That's a standalone issue.
The issue occurs because primary_group isn't set up within user create processor. It's set up only within user update processor. So function setUserGroups() in user create processor should be fixed.

**Comment by Lefthandmedia** [comment]
I guess you're right it is a separate issue: to recreate
- Create a new user  (give name and email address, do NOT save at this time)
- Go to permissions tab and add user to a group. When you save you can see the user ID is still set to 0 and the name to "none". thus this action has not been saved even though you clicked save
- Now when you go back to the 'general information' tab and save there. the newly chosen usergroup isn't send to the server for saving and thus does not get saved.

So the new and update action for user should also submit the data from the permissions tab

**Comment by wuuti** [comment]
Reviewing this #modxbughunt

**Comment by wuuti** [comment]
Can confirm this is still happening in current 2.5.x branch.

**Comment by sonicpunk** [comment]
#modxbughunt #1point to @wuuti

**Comment by Alroniks** [comment]
PR https://github.com/modxcms/revolution/pull/13528 merged

**Comment by mrhaw** [comment]
This fix should be ported to the Login.Register snippet. ...Currently we are forced to use a postHook
*&postHooks=`primarygroup`*
example:
```
$user = $hook->getValue('register.user');
$user->set("primary_group", 1);
$user->save();
````

No matter if we create users with Register or in code, it creates this error on user deletion, if not set:
*Error - table class/name for class: modAccess*
https://forums.modx.com/thread/?thread=96248
**(Spoke to soon on this - I still see this error sometimes)**

The extra getUserProfiles Snippet also looks to primary group of a user:
https://modx.com/extras/package/getuserprofiles

Wonder why this was changed? I will reference this url back here:
https://forums.modx.com/thread/88810/what-is-the-primary-group-in-the-users-table

**Comment by JoshuaLuckers** [comment]
Hi @mrhaw, thanks for the info. Is it possible to submit issues about this to the repo's of those extras?
- Login: https://github.com/modxcms/Login/issues

I can't find the repo for the `getuserprofiles` extra.

**Comment by mrhaw** [comment]
@JoshuaLuckers  I still see that error mentioned and now doubt any change regarding primary group is needed.
